### PR TITLE
Cut 0.0.101

### PR DESCRIPTION
**Pull request by TheBlueMatt**
First pass of CHANGELOG, though subject to change as more PRs are merged. Haven't done the serialization compat section yet.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,70 @@
...
+   new field `all_paths_failed` in `Event::PaymentFailed` is set to `false`.
+   This implies that the payment has not failed, but only one part. Payment
+   resolution is only indicated by an `Event::PaymentSent` event or an
+   `Event::PaymentFailed` with `all_paths_failed` set to `true`, which is also
+   set for the last remaining part of a multi-path payment (#1053).
+ * `lightning-invoice` now considers invoices with sub-millisatoshi precision
+   to be invalid, and requires millisatoshi values during construction (thus
+   you must call `amount_milli_satoshis` instead of `amount_pico_btc`, #1057).
+ * The `BaseSign` interface now includes two new hooks which provide additional
+   information about commigment transaction signatures and revocation secrets
```
Comment (ariard) [inline CHANGELOG.md:40]: s/commigment/commitment/g

**Review by ariard — COMMENTED** [review]
ACK 79fccde2

**Review by arik-so — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,92 @@
...
+ * Custom message types are now supported directly in the `PeerManager`,
+   allowing you to send and receive messages of any type that is not natively
+   understood by LDK. This requires a new type bound on `PeerManager`, a
+   `CustomMessageHandler`. `IgnoringMessageHandler` provides a simple default
+   for this new bound for ignoring unknown messages (#1031, #1074).
+ * Route graph updates as a result of failed payments are no longer provided as
+   `MessageSendEvent::PaymentFailureNetworkUpdate` but instead included in a
+   new field in the `Event::PaymentFailed` events. Generally, this means route
+   graph updates are no longer handled as a part of the `PeerManager` but
+   instead through the new `EventHandler` implementation of
```
Comment (jkczyz) [inline CHANGELOG.md:13]: nit: s/of/for

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,92 @@
...
+   events will be processed by the `NetGraphMsgHandler` prior to normal event
+   handling (#1043).
+ * The balances available on-chain to claim after a channel has been closed are
+   now exposed via `ChannelMonitor::get_claimable_balances` and
+   `ChainMonitor::get_claimable_balances`. The second can be used to get
+   information about all closed channels which still have on-chain balances
+   associated with them. See enum variants of `ln::channelmonitor::Balance` and
+   method documentation for the above methods for more information on the types
+   of balances exposed (#1034).
+ * When one HTLC out of several as a part of a multi-path payment fails, the
```
Comment (jkczyz) [inline CHANGELOG.md:29]: Maybe simply:

"When one HTLC of a multi-path payment fails"

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,92 @@
...
+   new field `all_paths_failed` in `Event::PaymentFailed` is set to `false`.
+   This implies that the payment has not failed, but only one part. Payment
+   resolution is only indicated by an `Event::PaymentSent` event or an
+   `Event::PaymentFailed` with `all_paths_failed` set to `true`, which is also
+   set for the last remaining part of a multi-path payment (#1053).
+ * To better capture the context described above, `Event::PaymentFailed` has
+   been renamed to `Event::PaymentPathFailed` (#1084).
+ * A new event, `ChannelClosed` is provided by `ChannelManager` when a channel
```
Comment (jkczyz) [inline CHANGELOG.md:37]: nit: comma after `ChannelClosed`

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,92 @@
...
+   provided by our counterparty, allowing additional verification (#1039).
+ * The `BaseSign` interface now includes additional information for cooperative
+   close transactions, making it easier for a signer to verify requests (#1064).
+ * `Route` has two additional helper methods to get fees and amounts (#1063).
+ * `Txid` and `Transaction` objects can now be deserialized from responses when
+   using the HTTP client in the `lightning-block-sync` crate (#1037, #1061).
+
+## Bug Fixes
+ * Fix a panic when reading a lightning invoice with a non-recoverable
+   signature. Further, restrict lightning invoice parsing require payment
```
Comment (jkczyz) [inline CHANGELOG.md:53]: s/require/to require

**Review by ariard — COMMENTED** [review]
ACK 414da222 modulo squashes!

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,92 @@
...
+ * Devrandom
+ * Galder Zamarreño
+ * Giles Cope
+ * Jeffrey Czyz
+ * Joseph Goulden
+ * Matt Corallo
+ * Sergi Delgado Segura
+ * Tibo-lg
+ * Valentine Wallace
+ * abhik-99
```
Comment (jkczyz) [inline CHANGELOG.md:99]: Abhik Banerjee

File: CHANGELOG.md [hunk 7]
```diff
@@ -65,11 +68,16 @@
    handle the long hangs in Bitcoin Core when it syncs to disk (#1090).
 
 ## Serialization Compatibility
- * 
- * XXX There should at least be an entry for considerations around balances
- * XXX There should at least be an entry for payment error graph updates
- * XXX THere should at least be an entry for duplicate paymentfailed/sent for MPPs sent on old versions
- * XXX probably more.
+ * Payments which were initiated in versions prior to 0.0.101 may still
```
Comment (jkczyz) [inline CHANGELOG.md:76]: s/which/that

(non-restrictive vs restrictive clause)

File: CHANGELOG.md [hunk 8]
```diff
@@ -65,11 +68,16 @@
...
+   generate duplicate `PaymentSent` `Event`s or may have spurious values for
+   `Event::PaymentPathFailed::all_paths_failed` (#1053).
+ * The return values of `ChannelMonitor::get_claimable_balances` (and, thus,
+   `ChainMonitor::get_claimable_balances`) may be spurious for channels where
+   the spend of the funding transaction appeared on chain while running a
+   version prior to 0.0.101. `Balance` information should only be relied upon
+   for channels which were closed while running 0.0.101+ (#1034).
```
Comment (jkczyz) [inline CHANGELOG.md:83]: s/which/that

File: CHANGELOG.md [hunk 9]
```diff
@@ -16,6 +16,9 @@
    contains an `Option`al `NetGraphmsgHandler`. If provided as `Some`, relevant
    events will be processed by the `NetGraphMsgHandler` prior to normal event
    handling (#1043).
+ * `NetworkGraph` is now, itself, thread-safe. Accordingly, most functions now
+   take `&self` instead of `&mut self` and the graph data can be accessed
+   through `NetworkGraph.read_only()` (#1043).
```
Comment (jkczyz) [inline CHANGELOG.md:21]: NetworkGraph::read_only

**Review by jkczyz — COMMENTED** [review]
Mostly nits

File: CHANGELOG.md (hunk 6 above)
Comment (TheBlueMatt) [reply CHANGELOG.md:99]: Hmm, where is that? The git author is `abhik-99`, which I tend to prefer to keep unless the author expresses otherwise.

Comment (TheBlueMatt) [reply CHANGELOG.md:99]: Hmm, where is that? The git author is `abhik-99`, which I tend to prefer to keep unless the author expresses otherwise.

Comment (jkczyz) [reply CHANGELOG.md:99]: Ah, I was looking at the corresponding GitHub page.

Comment (jkczyz) [reply CHANGELOG.md:99]: Let's use git-authors as it would be simpler to automate.

**Review by jkczyz — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Bumped the date and updated the change stats for the last-minute fixups. Will take after CI.

**Pull request merged**